### ISSUE TITLE
Replace timers.cpp implementation with std::chrono

### ISSUE THREAD

**Issue by rcurtin** [mlpack/mlpack#498]
Like the title says, this ticket can be resolved when the current code in `timers.cpp` and `timers.hpp` uses the support in `std::chrono` instead of the `timeval` struct and related code.  This will be more portable since it's part of the standard.

See #425 for more information and history.

**Comment by stereomatchingkiss** [comment]
Hi, I would like to take care of this issue.
Before I start, I have some questions to ask.

1 : std::chrono is very flexible, do you want to enhance original class?Like provide the users the options to choose between different clock and duration?
2 : If you agree with 1, maybe we could design new classes with template argument, use typedef to define the Timer and Timers classes(for backward compatibility)
3 : How to deal with timeval?Could we just change it to the rep type chrono?

**Comment by rcurtin** [comment]
Backward compatibility for timeval isn't really necessary, since the `CLI` class is the only one that actually prints the values of the timers.  Really the only need is that we have the functions

```
Timer::Start("timer_name");
Timer::Stop("timer_name");
Timer::Get("timer_name"); // the return type can change from timeval if needed
```

and that every timer that is run during a program is printed by `CLI::~CLI()`.  I don't see much need to provide any enhancements to the original class because only those three functions are ever used in the mlpack codebase and there's never been any need for more.

I guess maybe a better choice than the timeval class is the std::chrono::duration<double> class?  I'm indifferent to what's used here; the only thing that matters is that the output of `CLI::~CLI()` remains (just about) the same.  I'd suggest microseconds instead of milliseconds because that's the precision that's currently printed.

**Comment by na1taneja2821** [comment]
Hi,
Since the issue is still open, I would like to work on this. Also, since there is no backward compatibility requirement I will start with making changes to timers.hpp and timers.cpp

**Comment by rcurtin** [comment]
Fixed in #567.
